### PR TITLE
.Net: Update samples to use SK 0.18

### PR DESCRIPTION
**Pull request by glahaye**
### Motivation and Context
Time marches on,

### Description
Updated samples to use SK 0.18 and modified code when necessary due to that upgrade.

### Contribution Checklist
- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by shawncal** [comment]
https://github.com/microsoft/semantic-kernel/pull/2207 ??

File: samples/dotnet/KernelHttpServer/Extensions.cs [hunk 1]
```diff
@@ -132,7 +132,7 @@ internal static void RegisterNativeGraphSkills(this IKernel kernel, string graph
 
     internal static void RegisterTextMemory(this IKernel kernel)
     {
-        _ = kernel.ImportSkill(new TextMemorySkill(), nameof(TextMemorySkill));
+        _ = kernel.ImportSkill(nameof(TextMemorySkill));
```
Comment (shawncal) [inline samples/dotnet/KernelHttpServer/Extensions.cs:135]: This is just a string. Should be new TextMemorySkill, and pass in Memory.
See https://github.com/microsoft/semantic-kernel/pull/2207

**Review by shawncal — CHANGES_REQUESTED** [review]

**Comment by glahaye** [comment]
> #2207 ??

Argh. I just compared our branches. There are a couple of differences. Maybe merge one in then update the other to see what's missing?

File: samples/dotnet/KernelHttpServer/Extensions.cs (hunk 1 above)
Comment (glahaye) [reply samples/dotnet/KernelHttpServer/Extensions.cs]: Fixed

**Review by shawncal — APPROVED** [review]

**Pull request merged**